ta4j-core/src/main/java/org/ta4j/core/num/DoubleNum.java

```diff
@@ -187,8 +185,7 @@ public boolean equals(Object obj) {
         if(obj == NaN){
             return false;
         }
-        DoubleNum doubleNumObj = (DoubleNum) obj;
-        return Math.abs(delegate - doubleNumObj.delegate) < EPS;
+        return Double.compare(delegate, ((DoubleNum) obj).delegate) == 0 ? true : false;
```

Comment:
I'm concerned about having that `EPS` precision hidden away in `DoubleNum`.  I changed it and it did not impact any core unit tests.

Reply:
This is not the correct way to compare approximations (https://stackoverflow.com/questions/8081827/how-to-compare-two-double-values-in-java). For instance this will fail:
```
    @Test
    public void simpleEqualsTest(){
        Num a = numOf(1.000001);
        Num b = numOf(0.000001);
        assertNumEquals(numOf(1.0), a.minus(b));
    }

java.lang.AssertionError: 
Expected :1.0
Actual   :0.9999999999999999
 <Click to see difference>
```

By the way: you should skip `? true : false;` check